DeepLinkSDK/RouteHandler/DPLRouteHandler.h

```diff
@@ -51,6 +58,10 @@
  */
 - (UIViewController *)viewControllerForPresentingDeepLink:(DPLDeepLink *)deepLink;
 
+/**
+ returns the first `UINavigationController'
+ */
+- (UINavigationController *)navigationControllerForPresentingDeepLink:(DPLDeepLink *)deepLink;
```

Comment:
The proposal here makes the assumption that the application’s `rootViewController` is a a custom container view controller with a `UINavigationController` as a child view controller. While it is perfectly fine to set up an application with whatever structure makes sense for its use case, it’s not in the spirit of this library to make assumptions about the application’s view controller structure. For unique cases, the appropriate solution is to override `-viewControllerForPresentingDeepLink:` and return the desired view controller. 

Furthermore the method added here proposes a contract that it is the appropriate method to use to obtain a reference to a `UINavigationController` but it will not fulfill that contract if, for example, the application’s `rootViewController` is a `UINavigationController`. This unnecessarily complicates the interface.

I agree that it can potentially get redundant to specify the presenting view controller in each `DPLRouteHandler` subclass. Perhaps there should be a solution to define it one time if that's all the application calls for.